system/modules/core/library/Contao/StringUtil.php

```diff
@@ -491,13 +491,22 @@ public static function parseSimpleTokens($strString, $arrData)
 		// Replace the tags
 		foreach ($arrTags as $strTag)
 		{
-			if (strncmp($strTag, '{if', 3) === 0)
+			if (strncmp($strTag, '{if', 3) === 0 || strncmp($strTag, '{elseif', 7) === 0)
 			{
-				$strReturn .= preg_replace('/\{if ([A-Za-z0-9_]+)([=!<>]+)([^;$\(\)\[\]\}]+).*\}/i', '<?php if ($arrData[\'$1\'] $2 $3): ?>', $strTag);
-			}
-			elseif (strncmp($strTag, '{elseif', 7) === 0)
-			{
-				$strReturn .= preg_replace('/\{elseif ([A-Za-z0-9_]+)([=!<>]+)([^;$\(\)\[\]\}]+).*\}/i', '<?php elseif ($arrData[\'$1\'] $2 $3): ?>', $strTag);
+				$strReturn .= preg_replace_callback(
+					'/{(if|elseif) ([^=!<>]+)([=!<>]+)([^;$\\(\\)\\[\\]\\}]+).*\\}/i',
```

Comment:
It seems you have added more slashes than required in `\\(\\)\\[\\]\\}]+).*\\}`, haven't you?